examples/fit_no_covariates.py

```diff
@@ -0,0 +1,268 @@
+            "age_end": age_time[:, 0],
+            "year_start": age_time[:, 1],
+            "year_end": age_time[:, 1]
+        }))
+    return pd.concat(entries, ignore_index=True)
+
+
+def build_smoothing_grid(age, time):
+    """Builds a default smoothing grid with uniform priors."""
+    age_time = np.array(list(it.product(age["age"].values, time["year"].values)), dtype=np.float)
+    return pd.DataFrame({
+        "age": age_time[:, 0],
+        "year": age_time[:, 1],
+        "value_prior": "uniform01",
+        "age_difference_prior": "uniform",
+        "time_difference_prior": "uniform",
+        "const_value": np.NaN,
+    })
+
+
+def build_constraint(constraint):
+    """
+    This makes a smoothing grid where the mean value is set to a given
+    set of values.
+    """
+    return pd.DataFrame({
+        "age": constraint["age_start"],
+        "year": constraint["year_start"],
+        "value_prior": None,
+        "age_difference_prior": "uniform",
+        "time_difference_prior": "uniform",
+        "const_value": constraint["mean"],
+    })
+
+
+def internal_model(model_context, inputs):
+    config = model_context.parameters
+    model = model_context.input_data
+    # convert the observations to a normalized format.
+    model.observations = bundle_to_observations(config, inputs.observations)
+    model.constraints = bundle_to_observations(config, inputs.constraints)
+
+    rates_to_calculate_str = config.non_zero_rates.split()
+    age_df, time_df = age_year_from_data(inputs.constraints)
+    desired_outputs = integrand_outputs(
+        rates_to_calculate_str + ["prevalence"],
+        config.location_id, age_df, time_df)
+    model.outputs = desired_outputs
+
+    model.priors = pd.DataFrame({
+        "prior_name": ["uniform01", "uniform"],
+        "density_id": 0,  # uniform
+        "lower": [1e-10, np.NaN],
+        "upper": [1.0, np.NaN],
+        "mean": [0.01, 0.0],
+        "std": np.array([np.NaN, np.NaN], dtype=np.float),
+        "eta": np.array([np.NaN, np.NaN], dtype=np.float),
+        "nu": np.array([np.NaN, np.NaN], dtype=np.float),
+    })
+
+    smoothing_default = build_smoothing_grid(age_df, time_df)
+    # No smoothing for initial prevalence in Brad's example.
+    model.smoothers = {
+        "iota": smoothing_default,
+        "rho": smoothing_default,
+        "chi": smoothing_default,
+        "omega": build_constraint(model.constraints),
+    }
+    LOGGER.debug(f"Omega constraint {model.smoothers['omega']}")
+    return model_context
+
+
+def construct_database():
+    # Configuration
+    model_context = ModelContext()
+    model_context.parameters.bundle_id = 3209
+    model_context.parameters.tier_idx = 2
+    model_context.parameters.location_id = 26
+    model_context.parameters.non_zero_rates = "iota rho chi omega"
+
+    # Get the bundle and process it.
+    raw_inputs = data_from_csv(Path("measure.csv"))
+    model = internal_model(model_context, raw_inputs)
```

Comment:
looks like model is the same thing as model_context, doesn't look like it gets modified in internal_model, does it?

Reply:
No, it doesn't. Nicely noticed. I'm confused about what are results and what aren't. Something about this flow isn't working for me.